fluidkeys/secretreceive.go

```diff
@@ -87,25 +89,40 @@ func secretReceive() exitCode {
 			}
 			sawError = true
 		}
+		downloadedSecrets[key.Fingerprint()] = uuids
 	}
+
+	if len(downloadedSecrets) > 0 {
+		prompter := interactiveYesNoPrompter{}
+		out.Print("\n")
+		if prompter.promptYesNo("Delete now?", "Y", nil) == true {
+			for fingerprint, secretUUIDs := range downloadedSecrets {
+				for _, uuid := range secretUUIDs {
+					if err := client.DeleteSecret(fingerprint, uuid); err != nil {
+						log.Printf("failed to delete secret '%s': %v", uuid, err)
+					}
+				}
+			}
+		}
+	}
+
 	if sawError {
 		return 1
-	} else {
-		return 0
 	}
+	return 0
 }
 
-func downloadAndDecryptSecrets(key pgpkey.PgpKey) (decryptedSecrets []string, secretErrors []error, err error) {
+func downloadAndDecryptSecrets(key pgpkey.PgpKey) (decryptedSecrets []string, decryptedUUIDs []string, secretErrors []error, err error) {
```

Comment:
Nice for getting this working.

What do you think about this simpler alternative? return `secrets []secret, ...` where `secret` is:
```
type secret struct {
    decryptedContent string
    UUID uuid.UUID
    sentToFingerprint *fingerprint.Fingerprint
}
```